### PR TITLE
[Feat]Add the G304 rule example

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

### CODE REVIEW

File: docs/rules/g304_file-path_provided_as_taint_input.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+---
+id: g304
+title: G304: File path provided as taint input
+---
+
+Trying to open a file thtough a vatiable which can a dynamic or arbitrary files can be dangerous - exposing data, read triggering bugs etc. And if you're not absolutely sure it can never have use input, best to clean iy before use.
```
Comment (ccojocar) [inline docs/rules/g304_file-path_provided_as_taint_input.md]: typo: `vatiable` => `variable`

File: docs/rules/g304_file-path_provided_as_taint_input.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+## The right way
+
+```
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+    "path/filepath"
```
Comment (ccojocar) [inline docs/rules/g304_file-path_provided_as_taint_input.md]: small formatting issue

File: docs/rules/g304_file-path_provided_as_taint_input.md (hunk 1 above)
Comment (ccojocar) [reply docs/rules/g304_file-path_provided_as_taint_input.md]: I would rephrase the introduction as follows:

```
Trying to open a file provided as an input in a variable. The content of this variable might be controlled by an attacker who could change it to hold unauthorised file paths form the system.  In this way, it is possible to exfiltrate confidential information or such.

As a mitigation, it is recommended to sanitise  the file path before reading. For instance, you can check the file path against a whitelist with a set of allowed paths.
```

**Review by ccojocar — CHANGES_REQUESTED** [review]
@Aisuko Thanks for this doc contribution. It looks good. I would rephrase a bit the introduction. Please see my comment.

**Comment by Aisuko** [comment]
Done

**Review by ccojocar — COMMENTED** [review]
Thanks for fixing the review comments!

**Review by ccojocar — APPROVED** [review]
